src/azul/hmac.py

```diff
@@ -16,13 +17,16 @@ def verify(current_request):
     endpoint = f'{base_url}{path}'
     method = current_request.context['httpMethod']
     headers = current_request.headers
-    hmac_secret_key = config.hmac_key.encode()
+
+    def key_resolver(key_id, algorithm):
+        return aws.get_hmac_key_and_id_cached(key_id)['key']
```

Comment:
Uhh, this is may be a bit of a vulnerability. We definitely don't want to pass a user-supplied value to a function that extracts a secret using the user-supplied value.

Reply:
Ignore  this.